### PR TITLE
feat: add cohort selection [BD-38] [TNL-8713] [BB-4926]

### PR DESCRIPTION
**Pull request by Cup0fCoffee**
## Description
Add a field to post editor, which allows to select from course's cohorts.
"All learners" option is always available, and stands for no cohort.
screenshot

## Supporting information
[TNL-8713](https://openedx.atlassian.net/browse/TNL-8713)

## Testing steps
As discussions administrator:
- Create a post with a "All learners" cohort (i.e. either don't select anything under "Cohort visibility" field, or select "All Learners" option)
- Create a post with a cohort
- Check that can't edit cohort for an existing post

As normal user:
- Check that "Cohort visibility" field is not visible when trying to create or edit an existing post

### CODE REVIEW

File: src/discussions/posts/data/api.js [hunk 1]
```diff
@@ -73,14 +73,15 @@ export async function getThread(threadId) {
...
-export async function postThread(courseId, topicId, type, title, content, following = false) {
+export async function postThread(courseId, topicId, type, title, content, following = false, cohort) {
   const postData = snakeCaseObject({
     courseId,
     topicId,
     type,
     title,
     raw_body: content,
     following,
+    cohort,
```
Comment (xitij2000) [inline src/discussions/posts/data/api.js:84]:
```suggestion
    groupId: cohort,
```

Comment (xitij2000) [reply src/discussions/posts/data/api.js:84]: The name of the parameter in the API is group_id.

File: src/discussions/posts/data/api.js [hunk 2]
```diff
@@ -113,6 +114,7 @@ export async function updateThread(threadId, {
   following,
   closed,
   pinned,
+  cohort,
```
Comment (xitij2000) [inline src/discussions/posts/data/api.js:117]:
```suggestion

```

File: src/discussions/posts/post-editor/PostEditor.jsx [hunk 3]
```diff
@@ -94,6 +99,7 @@ function PostEditor({
         type: values.postType,
         title: values.title,
         content: values.comment,
+        cohort: values.cohort,
```
Comment (xitij2000) [inline src/discussions/posts/post-editor/PostEditor.jsx:102]:
```suggestion

```

File: src/discussions/posts/post-editor/PostEditor.jsx [hunk 4]
```diff
@@ -172,33 +180,55 @@ function PostEditor({
...
+                  as="select"
+                  value={values.cohort}
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  aria-describedby="cohortVisiblityInput"
+                  floatingLabel={intl.formatMessage(messages.cohortVisibility)}
+                >
+                  <option value="">{intl.formatMessage(messages.cohortVisibilityAllLearners)}</option>
+                  {cohorts.map(cohort => (
+                    <option key={cohort.id} value={cohort.groupId}>{cohort.name}</option>
```
Comment (xitij2000) [inline src/discussions/posts/post-editor/PostEditor.jsx:226]:
```suggestion
                    <option key={cohort.id} value={cohort.id}>{cohort.name}</option>
```

**Comment by Cup0fCoffee** [comment]
@xitij2000

Should there be validation for cohort in [Formic `validationSchema` field](https://github.com/edx/frontend-app-discussions/blob/eb9b4e57eaab6c7d76bb5112e9eb6227ce0f647b/src/discussions/posts/post-editor/PostEditor.jsx#L129-L141)?

File: src/discussions/posts/post-editor/PostEditor.jsx [hunk 5]
```diff
@@ -103,13 +108,15 @@ function PostEditor({
         title: values.title,
         content: values.comment,
         following: values.following,
+        cohort: authenticatedUser.administrator ? values.cohort : null,
```
[the remark this answers is not in the document]
Comment (xitij2000) [reply src/discussions/posts/post-editor/PostEditor.jsx:111]: I think perhaps we need to set this to undefined. I think setting it to null will pass along a null value.

**Comment by edx-webhook** [comment]
@Cup0fCoffee 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/frontend-app-discussions&entry.867055334=https://github.com/edx/frontend-app-discussions/pull/26&entry.1484655318=https://github.com/Cup0fCoffee&entry.752974735=2021-10-14+01:24&entry.1917517419=2021-11-02+06:38&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**